### ISSUE TITLE
Question: How to make ComputeSignature with silent: false show a PIN prompt?

### ISSUE THREAD

**Issue by heng-liu** [dotnet/runtime#51337]
### Description

I would like to check if there is a PIN prompt when ComputeSignature  with silent equals false. 
But seems there is no PIN prompt. I'm wondering if I missed anything? Thanks!

Here is the repro steps:
1. Git clone https://github.com/NuGet/Entropy.git
2. Create a self-issued .pfx with password by running 
`./Entropy/TestCertGenerator/CreateTestCertificate.ps1 -AddAsTrustedRootAuthority -Password password123`
3. Click 'YES' to add the certificate to trusted store
4. Create a console app targets .net5.0, with followings in Program.cs. 
(add package reference to `"System.Security.Cryptography.Pkcs" Version="5.0.1"`)
```
            string text = "xyz";
            byte[] bytes = Encoding.ASCII.GetBytes(text);
            ContentInfo contentInfo = new ContentInfo(bytes);
            var signedCms = new SignedCms(contentInfo);

            var cerPath = "<path to the .pfx file>";
            var cert = new X509Certificate2(cerPath, "password123");

            var cmsSigner = new CmsSigner(cert);

            signedCms.ComputeSignature(cmsSigner, silent: false);
```
5.Run the console app, no PIN prompt even if I set silent to false. And the signedCms is signed successfully.

May I know in which condition it will show a PIN prompt? Thanks.

### Configuration

Windows, net5.0

**Comment by bartonjs** [comment]
> May I know in which condition it will show a PIN prompt?

To get a PIN prompt it needs to be a private key that has a PIN.  That means either finding a Windows Hello PIN cert in X509Store(My, CurrentUser), or a smartcard, or when you import a PFX you need to check the "Enable strong private key protection" checkbox during import.

![image](https://user-images.githubusercontent.com/10642668/114951027-1e966880-9e09-11eb-8ceb-0a2fcc900d37.png)

You can programmatically import with X509KeyStorageFlags.UserProtected to do the same thing.  That should prompt for a PIN on import and then might also prompt when you use it... but it might not, since it might decide the PIN was entered in this process recently enough.

**Comment by heng-liu** [comment]
Thanks @bartonjs !
I tried to import the self-issued .pfx with "Enable strong private key protection" as following, but the above console app still doesn't have any PIN prompt with silent set to false.
I also tried to create a console targeting .net472, it doesn't prompt either..
I'm wondering if the password is cached somewhere and how shall I clear it? Or did I miss anything? Thanks!

![part1](https://user-images.githubusercontent.com/45407901/115082709-2c0c2b00-9ebb-11eb-804c-b676d9b900ad.png)
![part2](https://user-images.githubusercontent.com/45407901/115082852-6aa1e580-9ebb-11eb-93fe-74268d5b7356.png)
